src/components/filesystem/index.js

```diff
@@ -331,25 +332,32 @@ module.exports = {
                 that.currentDir = file.get();
                 that.updateFiles(refreshSharedWith);
             }).exceptionally(function(throwable) {
-                throwable.printStackTrace();
+                console.log(throwable.getMessage());
             });
         },
         updateFiles: function(refreshSharedWith) {
             var current = this.currentDir;
             if (current == null)
                 return Promise.resolve([]);
-            var that = this;
-            current.getChildren(that.getContext().crypto.hasher, that.getContext().network).thenApply(function(children){
-                var arr = children.toArray();
-                that.showSpinner = false;
-                that.files = arr.filter(function(f){
-                    return !f.getFileProperties().isHidden;
+            let that = this;
+            let context = this.getContext();
+            let directoryPath = peergos.client.PathUtils.directoryToPath(that.path);
+            context.getDirectorySharingState(directoryPath).thenApply(function(updatedSharedWithState) {
+                that.sharedWithState = updatedSharedWithState;
+                current.getChildren(that.getContext().crypto.hasher, context.network).thenApply(function(children){
+                    var arr = children.toArray();
+                    that.showSpinner = false;
+                    that.files = arr.filter(function(f){
+                        return !f.getFileProperties().isHidden;
+                    });
+                    if (refreshSharedWith) {
```

Comment:
when do we not want to refreshSharedWith?

Reply:
Simplified now

Reply:
Very nice